cpp/CMakeLists.txt

```diff
@@ -414,13 +414,22 @@ endfunction()
 set(Boost_DEBUG TRUE)
 set(Boost_USE_MULTITHREADED ON)
 set(Boost_USE_STATIC_LIBS ON)
-find_package(Boost COMPONENTS system filesystem REQUIRED)
-if ("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
-  set(BOOST_STATIC_SYSTEM_LIBRARY ${Boost_SYSTEM_LIBRARY_DEBUG})
-  set(BOOST_STATIC_FILESYSTEM_LIBRARY ${Boost_FILESYSTEM_LIBRARY_DEBUG})
+find_package(Boost COMPONENTS system filesystem)
+if (Boost_FOUND)
+  if ("${CMAKE_BUILD_TYPE}" STREQUAL "DEBUG")
+    set(BOOST_STATIC_SYSTEM_LIBRARY ${Boost_SYSTEM_LIBRARY_DEBUG})
+    set(BOOST_STATIC_FILESYSTEM_LIBRARY ${Boost_FILESYSTEM_LIBRARY_DEBUG})
+  else()
+    set(BOOST_STATIC_SYSTEM_LIBRARY ${Boost_SYSTEM_LIBRARY_RELEASE})
+    set(BOOST_STATIC_FILESYSTEM_LIBRARY ${Boost_FILESYSTEM_LIBRARY_RELEASE})
+  endif()
 else()
-  set(BOOST_STATIC_SYSTEM_LIBRARY ${Boost_SYSTEM_LIBRARY_RELEASE})
-  set(BOOST_STATIC_FILESYSTEM_LIBRARY ${Boost_FILESYSTEM_LIBRARY_RELEASE})
+  set(ARROW_BUILD_STATIC OFF)
+  set(ARROW_BUILD_TESTS OFF)
+  set(NO_TESTS ON)
+  set(ARROW_BUILD_UTILITIES OFF)
```

Comment:
We have the `ARROW_BOOST_USE_SHARED` CMake option https://github.com/apache/arrow/blob/master/cpp/CMakeLists.txt#L95 

the static libs are used in https://github.com/apache/arrow/blob/d28f1c1e0f21bc578b84ab4bed4cf259c333fbc9/cpp/src/arrow/io/CMakeLists.txt#L69 and in some of the unit tests -- perhaps instead if `ARROW_BOOST_USE_SHARED=on` then we should use the shared libraries for linking everywhere instead of the static ones?